src/main/scala/it/pagopa/interop/authorizationprocess/api/impl/ClientApiServiceImpl.scala

```diff
@@ -630,4 +630,27 @@ final case class ClientApiServiceImpl(
 
   }
 
+  override def removeArchivedPurpose(
+    purposeId: String
+  )(implicit contexts: Seq[(String, String)], toEntityMarshallerProblem: ToEntityMarshaller[Problem]): Route =
+    authorize(ADMIN_ROLE) {
+      val operationLabel: String = s"Removing archived Purpose $purposeId from all clients"
+      logger.info(operationLabel)
+
+      val result: Future[Unit] = for {
+        purposeUuid <- purposeId.toFutureUUID
+        purpose     <- purposeManagementService.getPurpose(purposeUuid)
+        _           <- purpose.versions
+          .find(_.state == PurposeManagementDependency.PurposeVersionState.ARCHIVED)
```

Comment:
a purpose can be considered ARCHIVED if all its versions are archived

Reply:
Please tell me What do you want to do in this case.